#inputs to main_node
int32 x
int32 y
int32 z

bool mode_select
bool button1
bool button2
bool button3